src/expression_evaluator/lambda_evaluator.cpp

```diff
@@ -72,6 +72,15 @@ void ListLambdaEvaluator::evaluate() {
     execFunc(params, *resultVector, &bindData);
 }
 
+bool ListLambdaEvaluator::selectInternal(common::SelectionVector& selVector) {
+    KU_ASSERT(children.size() == 1);
+    children[0]->evaluate();
+    validateListLen(params);
+    KU_ASSERT(resultVector->dataType.getLogicalTypeID() == common::LogicalTypeID::BOOL);
```

Comment:
what's if this assertion is triggered. e.g. using list_transform in WHERE

Reply:
If it is list_transform, a bidner error will be thrown.